operator/pkg/module/sync/runner_impl.go

```diff
@@ -156,7 +157,8 @@ func (r *runnerImpl) SyncModuleInfo(_ context.Context, kyma *v1alpha1.Kyma, modu
 			kyma.Status.ModuleInfos = append(kyma.Status.ModuleInfos, latestModuleInfo)
 		}
 	}
-	return updateRequired
+	statusUpdateRequiredFromDeletion := r.UpdateModuleInfos(ctx, kyma)
+	return updateRequired || statusUpdateRequiredFromDeletion
```

Comment:
I think we should avoid this or statements if possible, they are really hard to read. Is it possible to document this?

Reply:
I think this or statement is hard to avoid, the status updated determined by two parts, if the state of existing modules get changed or if module get deleted, I refactored this code again, please check if it's fine for you.